arch/xtensa/core/xtensa-asm2-util.S

```diff
@@ -52,22 +52,31 @@ xtensa_save_high_regs:
 	 * that means that the quad is owned by a wrapped-around call
 	 * in the registers, so we don't need to spill it or any
 	 * further registers from the GPRs and can skip to the end.
+	 *
+	 * Note that if GDB stub is enabled, all A4-A15 are being
+	 * stashed in stack for GDB to retrieve.
```

Comment:
I'm not completely understanding the point here.  Those other registers were never written by the current call, so any attempt by a debugger to read them is an error.  It's not just inspecting state, what's going to happen is that the access will trap to an exception to spill the registers instead (and... would that trap even work from within the debugger context?).

Basically: I don't see how this helps anything.  If you're in a state where this code is required to make something work, it seems like things must already be broken.

Isn't the correct behavior here for gdb to check WINDOWBASE/WINDOWSTART to ensure that the inspected register is correct and report some kind of runtime error instead?

Also, this changes the way the code behaves when gdbstub is enabled.  It's going to take longer to enter interrupts and exceptions and is going to use more stack doing it.  That seems undesirable.

Alternatively: if you absolutely can't avoid the need for gdb to read invalid registers, can't you just do a SPILL_ALL_WINDOWS somewhere in the interrupt context?  This would have the advantage of not changing the performance characteristics of the generated code.

Reply:
This was remnant of bits where I thought GDB needed all A0-A15 due to GDB reporting registers not available. Tried again without stashing all 16 registers and it is working now. So this is removed.